docs/guides/cloning-git-repositories.md

```diff
@@ -0,0 +1,69 @@
+# Cloning Git Repositories
+
+<div style="padding: 0px; margin: 0px;">
+  <span style="vertical-align:middle;">Author: </span>
+  <a href="https://github.com/BrunoQuaresma" style="text-decoration: none; color: inherit; margin-bottom: 0px;">
+    <span style="vertical-align:middle;">Bruno Quaresma</span>
+    <img src="https://avatars.githubusercontent.com/u/3165839?v=4" width="24px" height="24px" style="vertical-align:middle; margin: 0px;"/>
+  </a>
+</div>
+August 06, 2024
+
+---
+
+When starting to work on a project, engineers usually need to clone a Git
+repository. Even though this is often a quick step, it can be automated using
+the [Coder Registry](https://registry.coder.com/) to make a seamless Git-first
+workflow.
+
+The first step to enable Coder to clone a repository is to provide
+authorization. This can be achieved by using the Git provider, such as GitHub,
+as an authentication method. If you don't know how to do that, we have written
+documentation to help you:
+[Authentication with GitHub](https://coder.com/docs/admin/auth#github).
+
```

Comment:
Can we add a note on supporting Gitlab as well? We don't need to copy the setup for Gitlab and self hosted git, just mention that they are alternative options. 

https://coder.com/docs/admin/external-auth#gitlab-self-managed
https://coder.com/docs/admin/external-auth#self-managed-git-providers

It's discussed in the Git Clone module README as well.